test/telemetry/src/main/java/org/opensearch/test/telemetry/tracing/SpanDataValidator.java

```diff
@@ -0,0 +1,25 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.test.telemetry.tracing;
+
+import org.opensearch.telemetry.tracing.Span;
+
+import java.util.List;
+
+/**
+ * Performs validations on span Data.
+ */
+public interface SpanDataValidator {
+    /**
+     * Validates spanData and return boolean value on validation status.
+     * @param spans spans emitted at any point of time.
+     * @param requests requests can be search/index calls.
+     */
+    public boolean validate(List<MockSpanData> spans, int requests);
```

Comment:
Why do we need `requests`?

Reply:
This is needed for validations specific to traceIDs and parent spans, as number of requests will determine these numbers.